### PR TITLE
[mem] Modify macro definitions for better code reading

### PR DESCRIPTION
**Pull request by wdfk-prog**
## 拉取/合并请求描述：(PR description)

[
#### 为什么提交这份PR (why to submit this PR)
- https://github.com/RT-Thread/rt-thread/issues/8761

#### 请提供验证的bsp和config (provide the config and bsp) 
- BSP: bsp/stm32/stm32h750-artpi
- .config:
CONFIG_RT_USING_SMALL_MEM=y
CONFIG_RT_USING_SMALL_MEM_AS_HEAP=y
PR branch:
- https://github.com/wdfk-prog/ART-PI/commit/8bc4f2cee1b02a9236c6d0c77acd83e2dde5b5e0

#### 没有在64位上验证,无测试条件
]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [x] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 代码是高质量的 Code in this PR is of high quality
- [ ] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

**Comment by milesight-dept2** [comment]
```
  remotes/origin/master
  remotes/origin/serialX-dev
  remotes/origin/stable-v1.2.x
  remotes/origin/stable-v2.0.x
  remotes/origin/stable-v2.1.x
  remotes/origin/stable-v3.0.x
  remotes/origin/v4.0.x
  remotes/origin/v5.0.x
  remotes/origin/v5.1.x
  remotes/pull/8762/merge
[format_ignore.py 79 INFO] modified file -> src/mem.c 
[cpp_check.py 23 INFO] Start to static code analysis. 
[cpp_check.py 44 INFO] Checking src/mem.c ...
Checking src/mem.c: RT_ASSERT(x)=;rt_list_for_each_entry(a,b,c)=a=(void*)b;...
Checking src/mem.c: RT_ASSERT(x)=;rt_list_for_each_entry(a,b,c)=a=(void*)b;;ARCH_CPU_64BIT;RT_USING_MEMTRACE;RT_USING_SMALL_MEM...
Checking src/mem.c: RT_ASSERT(x)=;rt_list_for_each_entry(a,b,c)=a=(void*)b;;ARCH_CPU_64BIT;RT_USING_SMALL_MEM...
Checking src/mem.c: RT_ASSERT(x)=;rt_list_for_each_entry(a,b,c)=a=(void*)b;;RT_USING_FINSH;RT_USING_MEMTRACE;RT_USING_SMALL_MEM...
Checking src/mem.c: RT_ASSERT(x)=;rt_list_for_each_entry(a,b,c)=a=(void*)b;;RT_USING_FINSH;RT_USING_SMALL_MEM...
Checking src/mem.c: RT_ASSERT(x)=;rt_list_for_each_entry(a,b,c)=a=(void*)b;;RT_USING_MEMTRACE;RT_USING_SMALL_MEM...
Checking src/mem.c: RT_ASSERT(x)=;rt_list_for_each_entry(a,b,c)=a=(void*)b;;RT_USING_SMALL_MEM...
 
[cpp_check.py 45 INFO] src/mem.c:562:5: error: Uninitialized variable: name [uninitvar]
    name = argc > 1 ? argv[1] : RT_NULL;
    ^
src/mem.c:609:5: error: Uninitialized variable: name [uninitvar]
    name = argc > 1 ? argv[1] : RT_NULL;
    ^
 
[cpp_check.py 73 ERROR] static code analysis(cppcheck) fail. 
Error: Process completed with exit code 1.
```

- 出错的地方我查看没问题,CI出问题了吧

**Comment by wdfk-prog** [comment]
```log
[cpp_check.py 23 INFO] Start to static code analysis. 
[cpp_check.py 44 INFO] Checking src/mem.c ...
Checking src/mem.c: RT_ASSERT(x)=;rt_list_for_each_entry(a,b,c)=a=(void*)b;...
Checking src/mem.c: RT_ASSERT(x)=;rt_list_for_each_entry(a,b,c)=a=(void*)b;;ARCH_CPU_64BIT;RT_USING_MEMTRACE;RT_USING_SMALL_MEM...
Checking src/mem.c: RT_ASSERT(x)=;rt_list_for_each_entry(a,b,c)=a=(void*)b;;ARCH_CPU_64BIT;RT_USING_SMALL_MEM...
Checking src/mem.c: RT_ASSERT(x)=;rt_list_for_each_entry(a,b,c)=a=(void*)b;;RT_USING_FINSH;RT_USING_MEMTRACE;RT_USING_SMALL_MEM...
Checking src/mem.c: RT_ASSERT(x)=;rt_list_for_each_entry(a,b,c)=a=(void*)b;;RT_USING_FINSH;RT_USING_SMALL_MEM...
Checking src/mem.c: RT_ASSERT(x)=;rt_list_for_each_entry(a,b,c)=a=(void*)b;;RT_USING_MEMTRACE;RT_USING_SMALL_MEM...
Checking src/mem.c: RT_ASSERT(x)=;rt_list_for_each_entry(a,b,c)=a=(void*)b;;RT_USING_SMALL_MEM...
 
[cpp_check.py 45 INFO] src/mem.c:562:5: error: Uninitialized variable: name [uninitvar]
    name = argc > 1 ? argv[1] : RT_NULL;
    ^
src/mem.c:609:5: error: Uninitialized variable: name [uninitvar]
    name = argc > 1 ? argv[1] : RT_NULL;
    ^
 
[cpp_check.py 73 ERROR] static code analysis(cppcheck) fail.
```

- 我看了错误地方是没问题的;CI出问题了吧

File: src/mem.c [hunk 1]
```diff
@@ -87,16 +87,12 @@ struct rt_small_mem
...
-#define MIN_SIZE 12
-#endif /* ARCH_CPU_64BIT */
+#define MIN_SIZE (sizeof(rt_ubase_t) + sizeof(rt_size_t) + sizeof(rt_size_t))
 
 #define MEM_MASK ((~(rt_size_t)0) - 1)
 
-#define MEM_USED()         ((((rt_base_t)(small_mem)) & MEM_MASK) | 0x1)
-#define MEM_FREED()        ((((rt_base_t)(small_mem)) & MEM_MASK) | 0x0)
+#define MEM_USED(_mem)       ((((rt_base_t)(_mem)) & MEM_MASK) | 0x1)
+#define MEM_FREED(_mem)      ((((rt_base_t)(_mem)) & MEM_MASK) | 0x0)
```
Comment (mysterywolf) [inline src/mem.c:95]: 就用mem就好 _mem看着有些难看

**Comment by mysterywolf** [comment]
CI 报错不用关心

File: src/mem.c (hunk 1 above)
Comment (wdfk-prog) [reply src/mem.c:95]: image

其他宏是这么命名的,我只是保持统一

Comment (mysterywolf) [reply src/mem.c:95]: 都去掉下划线吧 挺丑的，内部函数下划线开头是对的 变量没有啥必要

**Review by mysterywolf — APPROVED** [review]

**Pull request merged**